src/core/include/openvino/runtime/tensor_util.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <filesystem>
+
+#include "openvino/core/partial_shape.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+
+OPENVINO_API
+Tensor create_mmaped_tensor(const Tensor& tensor, const std::string& file_name);
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
+OPENVINO_API
+Tensor create_mmaped_tensor(const Tensor& tensor, const std::wstring& file_name);
+#endif
+
+#ifdef OPENVINO_CPP_VER_AT_LEAST_17
+OPENVINO_API
+Tensor create_mmaped_tensor(const Tensor& tensor, const std::filesystem::path& file_name);
```

Comment:
My initial PR covered use cases beyond offloading tensor content to a file. For example, reading a file's content as a memory mapped Tensor content is also required for memory-efficient LoRA adapter loading. Now, I don't see that kind of API here. @iefode, FYI.

Reply:
added